compress.cc

```diff
@@ -32,6 +32,16 @@ class lz4_processor: public compressor {
     size_t compress(const char* input, size_t input_len, char* output,
                     size_t output_len) const override;
     size_t compress_max_size(size_t input_len) const override;
+
+    // Note: lz4 compressor does _not_ implement the scatterered
+    // compress/decompress API:s. This is on purpose, not just me
+    // being lazy. The reason why is that lz4 does not have an API
+    // to do incremental comp/decomp. The closest we get is block-incremental,
+    // but this requires symmetric comp/decomp (i.e. must store compressed
+    // bounds per block in a "stream"), or using "frame" API (lzframe).
+    // Neither of these would create an end result that is the same
+    // as the above, non-scattered, API calls. So for lz4 we just use
+    // the linearized fallback. Or in other words: don't use lz4 for this.
```

Comment:
That's a pity, since lz4 is the fastest compressor out there.

Reply:
To a large extent by requiring a lot of pre-conditions for running, which are somewhat orthogonal to what we are doing here... :-P

Reply:
I think it's mostly due to a good implementation and due to compromising on the  compression factor.